### ISSUE TITLE
current make_const is a bit confusing

### ISSUE THREAD

**Issue by pengwa** [onnx/tensorflow-onnx#128]
all make_const usage is like below:

```
onnx_node = g.make_const(..., node_name)
.. some other node reference 'node_name'
.. onnx_node is never used. 
```

So this bring confusion, when we call make_const, why we don't add the created onnx_node into final graph. The reason is also simple: when we say create a const node, we actually are assuming that we want make the constant data as an "initializer" in onnx node, which is part of make_const did. 

I think to avoid the confusion, we can just make the func make_node behaviour be clear --- just add the const as an initialize,  there won't be extra node added or created or returned. 

@guschmue  please kindly let me you your thoughts on this. I can provide a PR here. 

By the way, share confusions i got during code. 
When I create a ONNX node in converter logic, we user helper.make_node(), and need append the returned node into a set of ops, which will be used in g.set_nodes(ops). 
While when I create constant named "a", I use make_const, I CANT do the same thing, otherwise, there will be name conflicting because I add "a" as a initializer, and as a constant node , in final onnx graph. 

I believe we could do some refinement here to avoid this confusion.

**Comment by pengwa** [comment]
https://github.com/onnx/tensorflow-onnx/pull/129 for your suggestions.

**Comment by guschmue** [comment]
hm, yes - that was intentional to have that node created. 
It simplifies things:
say you look at node.input[x] for example and that node is actually an initializer. Since the cost node is a object you can follow that input, get to the node, get the tensor value ... all the same as normal nodes and const nodes that came from tensorflow (because they look exactly like this).
The confusing part is that the const node is not in the onnx node list. Same is true for const nodes that came from tensorflow.

**Comment by guschmue** [comment]
ah, the node = ctx.make_const() ... yes, we could remove the assign part.

**Comment by pengwa** [comment]
right!!! got it!

So maybe I can just remove the returned value of make_node, in case we added it again by mistake?

**Comment by guschmue** [comment]
yes, that is a good idea. And passing in "Const" doesn't add much value either.

**Comment by guschmue** [comment]
btw, the const that come down from tensorflow do the same thing as make_const ... by returning None the node gets out of the node list but the object stays around and remains in the name to node dict:
https://github.com/onnx/tensorflow-onnx/blob/master/tf2onnx/tfonnx.py#L217

**Comment by pengwa** [comment]
yes, I noticed that. :)
